include/swift/AST/ReferencedNameTracker.h

```diff
@@ -21,6 +21,15 @@ namespace swift {
 class NominalTypeDecl;
 
 class ReferencedNameTracker {
+  StringRef ReferenceDependenciesFilePath;
+
+public:
+  ReferencedNameTracker(StringRef referenceDependenciesFilePath)
```

Comment:
Nitpick: please mark this `explicit` to avoid accidental implicit conversions.

(Even though that's really unlikely in this case, it's just a reasonable habit for any one-argument constructor.)

Reply:
More importantly, I feel a little weird about putting output paths into the AST layer. I feel like that logic should be path-agnostic, and it should be the Frontend that deals with where things go. But does that result in us passing around much more information?

Reply:
Fixing the nitpick was the easy part:)

Addressing output paths in the AST layer: 
In the Frontend layer: emitReferenceDependencies is passed a SourceFile, from which it gets the tracker, from which it gets the path for writing the info.
But the tracker is in the AST layer, as is the SourceFile.
Hmm.... since emitReferenceDependencies is called from emitReferenceDependenciesIfNeeded in frontendTool.cpp, an alternative design would be to pass the required output path down as an argument through those calls. Does that design fit the layering better?

Reply:
That's what I like, although it does require keeping the SourceFile in sync with its SupplementaryOutputPaths. But that's the same as what we had to do with PrimarySpecificPaths, so maybe that's fine.

Reply:
Come to think of it, since emitReferenceDependencies gets FrontendOptions, it ought to be able to do the lookup. I'll explore that.

Reply:
Well, how to keep it in sync? Could replace the PrimarySourceFiles vector with a vector of pairs of Source and Input files. Could implement a hash map input name to input filename in FrontendInputsAndOutputs. Any other ideas?